packages/react-native-sdk/docusaurus/docs/reactnative/05-ui-cookbook/21-closed-captions.mdx

```diff
@@ -0,0 +1,103 @@
+---
+id: closed-captions
+title: Closed Captions
+description: How to add closed captions to your calls
+---
+
+The Stream API supports adding real-time closed captions (subtitles for participants) to your calls.
+This guide shows you how you can build it on top of our React Native SDK.
+
+## Prerequisites
+
+Make sure that the closed caption feature is enabled in your app's dashboard.
+The closed caption feature can be set on the call type level, and the available options are:
+
+- `available`: the feature is available for your call and can be enabled.
+- `disabled`: the feature is not available for your call. In this case, it's a good idea to "hide" any UI element you have related to closed captions.
+- `auto-on`: the feature is available and will be enabled automatically once the user is connected to the call.
+
+It's also possible to override the call type's default when creating a call:
+
+```ts
+await call.getOrCreate({
+  data: {
+    settings_override: {
+      transcription: {
+        mode: 'available',
+        closed_caption_mode: 'available',
+      },
+    },
+  },
+});
+```
+
+For the purposes of this article, make sure that closed captions are either `available` or `auto-on`.
+
+## Enabling, disabling and tweaking closed captions
+
+```tsx
+await call.startClosedCaptions(); // enable closed captions
+await call.stopClosedCaptions(); // disable closed captions
+
+call.updateClosedCaptionSettings({
+  retentionTimeInMs: 2700, // the duration a caption can stay in the queue
+  queueSize: 2, // number of captions that can be stored in the queue
+});
+```
+
+## Check if closed captions are enabled
+
+```tsx
+import { useCallStateHooks } from '@stream-io/video-react-native-sdk';
+
+const { useIsCallCaptioningInProgress } = useCallStateHooks();
+const isCaptioningInProgress = useIsCallCaptioningInProgress();
+
+console.log(
+  `Closed captions are ${isCaptioningInProgress ? 'enabled' : 'disabled'}`
+);
+```
+
+## Rendering closed captions
+
+Once enabled, the SDK will expose the closed caption events to your application via a special `useCallClosedCaptions` hook.
+This is how it can be used:
+
+```tsx
+import { useCallStateHooks } from '@stream-io/video-react-native-sdk';
+
+export const ClosedCaptions = () => {
+  const { useCallClosedCaptions } = useCallStateHooks();
+  const closedCaptions = useCallClosedCaptions();
+  return (
+    <View style={styles.rootContainer}>
+      {closedCaptions.map(({ speaker_name, text }) => (
+        <View style={styles.closedCaptionItem} key={`${speaker_name}/${text}`}>
```

Comment:
Let's use speaker name + start time as key, since that's what we deduplicate on. Otherwise this key may not be unique.

Reply:
Fixed with: [6901fda](https://github.com/GetStream/stream-video-js/pull/1508/commits/6901fda8bbe93dcbe934902eeca9a6b3eb625a67)